clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java

```diff
@@ -17,90 +17,384 @@
-    PassthroughServer newServer = new PassthroughServer(true);
-    newServer.registerServerEntityService(new EhcacheServerEntityService());
-    newServer.registerClientEntityService(new EhcacheClientEntityService());
-    newServer.registerServerEntityService(new CoordinationServerEntityService());
-    newServer.registerClientEntityService(new ClientCoordinationEntityService());
+  private static Logger LOGGER = LoggerFactory.getLogger(UnitTestConnectionService.class);
 
-    /*
-     * Construct an off-heap resource configuration and register an OffHeapResourcesProvider.
-     */
-    OffheapResourcesType resources = new OffheapResourcesType();
-    List<ResourceType> resourcesList = resources.getResource();
-    resourcesList.add(defineResource("defaultResource", MemoryUnit.MB, 128L));
-    resourcesList.add(defineResource("primary-server-resource", MemoryUnit.MB, 64L));
-    resourcesList.add(defineResource("secondary-server-resource", MemoryUnit.MB, 64L));
-    OffHeapResourcesConfiguration ohrpConfig = new OffHeapResourcesConfiguration(resources);
-    newServer.registerServiceProvider(new OffHeapResourcesProvider(), ohrpConfig);
-
-    newServer.start();
-    return newServer;
+  private static final Map<URI, ServerDescriptor> servers = new HashMap<URI, ServerDescriptor>();
+
+  /**
+   * Adds a {@link PassthroughServer} if, and only if, a mapping for the {@code URI} supplied does not
+   * already exist.  The server is started as it is added.
+   *
+   * @param uri the {@code URI} for the server; only the <i>scheme</i>, <i>host</i>, and <i>port</i>
+   *            contribute to the server identification
+   * @param server the {@code PassthroughServer} instance to use for connections to {@code uri}
+   */
+  public static void add(URI uri, PassthroughServer server) {
+    URI keyURI = createKey(uri);
+    if (servers.containsKey(keyURI)) {
+      throw new AssertionError("Server at " + uri + " already provided; use remove() to remove");
+    }
+
+    servers.put(keyURI, new ServerDescriptor(server));
+    server.start();
+    LOGGER.info("Started PassthroughServer at {}", keyURI);
   }
 
-  private static ResourceType defineResource(String name, MemoryUnit unit, long size) {
-    final ResourceType resource = new ResourceType();
-    resource.setName(name);
-    resource.setUnit(unit);
-    resource.setValue(BigInteger.valueOf(size));
-    return resource;
+  /**
+   * Adds a {@link PassthroughServer} if, and only if, a mapping for the URI supplied does not
+   * already exist.  The server is started as it is added.
+   *
+   * @param uri the URI, in string form, for the server; only the <i>scheme</i>, <i>host</i>, and <i>port</i>
+   *            contribute to the server identification
+   * @param server the {@code PassthroughServer} instance to use for connections to {@code uri}
+   */
+  public static void add(String uri, PassthroughServer server) {
+    add(URI.create(uri), server);
```

Comment:
This should really be `new URI(uri)` if we're going to have this method shouldn't it?